test/k8sT/Updates.go

```diff
@@ -303,22 +292,28 @@ func InstallAndValidateCiliumUpgrades(kubectl *helpers.Kubectl, oldVersion, newV
 		}
 
 		By("Install Cilium pre-flight check DaemonSet")
-		err = kubectl.CiliumPreFlightInstall(helpers.CiliumDefaultPreFlightPatch)
-		ExpectWithOffset(1, err).To(BeNil(), "Cilium pre-flight %q was not able to be deployed", newVersion)
+
+		res = kubectl.ExecMiddle("helm template " +
+			helpers.HelmTemplate + " " +
+			"--namespace=kube-system " +
+			"--set preflight.enabled=true " +
+			"--set agent.enabled=false " +
+			"--set config.enabled=false " +
+			"--set operator.enabled=false " +
+			"> cilium-preflight.yaml")
+		ExpectWithOffset(1, res).To(helpers.CMDSuccess(), "Unable to generate preflight YAML")
+
+		res = kubectl.Apply("cilium-preflight.yaml")
+		ExpectWithOffset(1, res).To(helpers.CMDSuccess(), "Unable to deploy preflight manifest")
 		ExpectCiliumPreFlightInstallReady(kubectl)
 
 		// Once they are installed we can remove it
 		By("Removing Cilium pre-flight check DaemonSet")
-		kubectl.Delete(helpers.GetK8sDescriptor(helpers.CiliumDefaultPreFlight))
+		kubectl.Delete("cilium-preflight.yaml")
 
-		cmPatch := helpers.CiliumConfigMapPatch
-		err = kubectl.CiliumInstall(helpers.CiliumDefaultDSPatch, cmPatch)
+		err = kubectl.CiliumInstall([]string{})
```

Comment:
We need to make sure we are using the kvstore-based identity allocator here - see https://github.com/cilium/cilium/pull/8734/files .

Reply:
Yes! This explains the test failure as well.